src/lib.rs

```diff
@@ -39,7 +39,7 @@ impl Cid {
         Cid {
             version,
             codec,
-            hash: hash.into(),
+            hash: Multihash::from_bytes(hash.into()).unwrap(),
```

Comment:
This will now panic when hash contains an invalid multihash. Before you could pass just anything. Not sure what to do about this.